methods.rb

```diff
@@ -0,0 +1,116 @@
+module Enumerable
+  def my_each
+    i = 0
+    length.times do
+      yield(self[i])
+      i += 1
+    end
+    self
+  end
+
+  def my_map(proc = nil)
+    temp = []
+    i = 0
+    if proc.nil?
+      length.times do
+        temp.push(yield(self[i]))
+        i += 1
+      end
+    else
+      length.times do
+        temp.push(proc.call(self[i]))
+        i += 1
+      end
+    end
+    temp
+  end
+
+  def my_each_with_index
+    i = 0
+    length.times do
+      yield(self[i], i)
+      i += 1
+    end
+    self
+  end
+
+  def my_select
+    temp = []
+    i = 0
+    length.times do
+      temp.push(self[i]) if yield(self[i])
+      i += 1
+    end
+    temp
+  end
+
+  def my_all?
+    i = 0
+    length.times do
+      return false unless yield(self[i])
+
+      i += 1
+    end
+    true
+  end
+
+  def my_any?
+    i = 0
+    length.times do
+      return true if yield(self[i])
+
+      i += 1
+    end
+    false
+  end
```

Comment:
- [ ] `my_any` is not working as the original `any?` method, also is not passing all test cases. 

### `any?` is an Enumerable that has these characteristics:
- Returns `true` if at least one element is not false.
```
[1, false, nil].any?
# true
```
- Returns `false` if all elements are false.
```
[nil, false, nil].any?
# false
```
- it doesn't change the original array/range/hash
- it can accept as arguments a Class, a RegEx, or a pattern, and returns true or false depending if the values in the array/range/hash match the condition
- it can use a block but it is not necessary to make the method work

[Enumerable#any?](https://ruby-doc.org/core-2.7.2/Enumerable.html#method-i-any-3F)